Doesn't compile with boost-1.71.

**Describe the bug**
The build sript `build.sh` doesn't find Boost's component `boost_program_options`

**To Reproduce**
Steps to reproduce the behavior:
1. Run command `sh build.sh -t local`

**Expected behavior**
The build script run cmake buildling without errors.

**Logs**
```
-- Eclipse version is set to 3.6 (Helios). Adjust CMAKE_ECLIPSE_VERSION if this is wrong.
-- Version info
--   product version: 4.0.0
--   git commit: d86b6eefef1437c34bee459bd24c573b32cb4f47
--   git branch: Release4
-- Dependency info
--   boost version: 1.65
--   openssl version: 1.0.0 ()
-- System info
--   host system name: Linux
--   host system processor: x86_64
--   system processor: x86_64
--   system name: Linux
--   host system: Linux-5.4.0-42-generic
CMake Error at /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake:117 (find_package):
  Found package configuration file:

    /usr/lib/x86_64-linux-gnu/cmake/boost_program_options-1.71.0/boost_program_options-config.cmake

  but it set boost_program_options_FOUND to FALSE so package
  "boost_program_options" is considered to be NOT FOUND.  Reason given by
  package:

  No suitable build variant has been found.

  The following variants have been tried and rejected:

  * libboost_program_options.so.1.71.0 (release, Boost_USE_RELEASE_LIBS=OFF)

  * libboost_program_options.a (static, default is shared, set
  Boost_USE_STATIC_LIBS=ON to override)

Call Stack (most recent call first):
  /usr/lib/x86_64-linux-gnu/cmake/Boost-1.71.0/BoostConfig.cmake:182 (boost_find_component)
  /usr/share/cmake-3.16/Modules/FindBoost.cmake:443 (find_package)
  CMakeLists.txt:140 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/flipback/projects/ASNeG/OpcUaStack/build_local_Debug/CMakeFiles/CMakeOutput.log".

```

**Environment (please complete the following information):**
 - Ubunt 20.04
 - Release4 branch

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -24,6 +24,7 @@
 
 **Bugs**:
 
+* fix compiltation for boost 1.70 and higher #428, by @flipback
 * fix reconnect timer crash in session service #302, by @huebl
 * Add isNull and setNull function to class OpcUaNodeId #278, @huebl
 * Add Generator directory to cmake file #272, @huebl

--- a/docs/source/1_getting_started/installation.rst
+++ b/docs/source/1_getting_started/installation.rst
@@ -13,7 +13,7 @@ the following requirements:
 
 * Compiler with C++17 support (we test only GCC, Clang and Visual Studio)
 * CMake >= 3.5
-* Boost >= 1.58
+* Boost >= 1.65
 * OpenSSL >= 1.0.0
 
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -137,9 +137,9 @@ message(STATUS "  host system: ${CMAKE_HOST_SYSTEM}")
     set(Boost_USE_RELEASE_LIBS   ON)
  endif()
 add_definitions(-DBOOST_ALL_DYN_LINK)
+set(Boost_NO_BOOST_CMAKE ON)
 find_package(
     Boost 
-    "${BOOST_VERSION_MAJOR}.${BOOST_VERSION_MINOR}" 
     COMPONENTS program_options system unit_test_framework filesystem thread date_time chrono regex 
     REQUIRED
 )

